pkg/genericapiserver/genericapiserver.go

```diff
@@ -475,6 +477,14 @@ func (s *GenericAPIServer) init(c *Config) {
 
 	attributeGetter := apiserver.NewRequestAttributeGetter(s.RequestContextMapper, s.NewRequestInfoResolver())
 	handler = apiserver.WithAuthorizationCheck(handler, attributeGetter, s.authorizer)
+	if len(c.AuditLog) != 0 {
+		auditLog, err := os.OpenFile(c.AuditLog, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
```

Comment:
In contrib/mesos (https://github.com/kubernetes/kubernetes/blob/master/contrib/mesos/pkg/minion/server.go#L226) we use lumberjack.v2 for log rotation. Maybe that's an option here as well.